src/test/java/com/williamfiset/algorithms/datastructures/balancedtree/RedBlackTreeTest.java

```diff
@@ -246,6 +246,32 @@ public void testRandomizedValueInsertionsAgainstTreeSet() {
     }
   }
 
+  @Test
+  public void testRemoval() {
+    tree.insert(5);
+    tree.insert(7);
+    tree.insert(9);
+
+    tree.delete(5);
+    assertFalse(tree.contains(5));
+
+    tree.delete(7);
+    assertFalse(tree.contains(7));
+
+    tree.delete(9);
+    assertFalse(tree.contains(9));
+  }
+
+  @Test
+  public void testNullRemoval() {
+    assertFalse(tree.delete(null));
+  }
+
+  @Test
+  public void testNumberDoesntExist() {
+    assertFalse(tree.delete(0));
+  }
+
   /*
     @Test
     public void randomRemoveTests() {
```

Comment:
Can you uncomment the randomTemoveTests method? You may need to adapt it to work, but it should catch any bugs

Reply:
Oh, sorry about that. By the way (this is not a hate comment, just a suggestion), it would much easier if instead of null checks, you used a constant field called NIL or something which is a node with the color of black, the left child is NIL, the right child is NIL, and the parent is NIL. So, instead of constantly doing if(node == null || node.color == BLACK) you can do if(node.color == BLACK). Like I said, just a suggestion. I didn't change it because you were the original author.